mlrun/feature_store/steps.py

```diff
@@ -226,34 +227,96 @@ def _do_pandas(self, event):
+                    else:
+                        column_filter = ~isnull(col(column))
+                    if isinstance(new_col_type, (FloatType, DoubleType, DecimalType)):
+                        new_column_filter = isnull(col(new_column_name)) | isnan(
+                            col(new_column_name)
+                        )
+                    else:
+                        #  we need to check that every value replaced if we changed column type - except None or NaN.
+                        new_column_filter = isnull(col(new_column_name))
+                    mapping_to_null = [
+                        k
+                        for k, v in column_map.items()
+                        if v is None
+                        or (
+                            isinstance(v, (float, np.float64, np.float32, np.float16))
+                            and math.isnan(v)
+                        )
+                    ]
+                    turned_to_none_values = df.filter(
+                        (column_filter) & (new_column_filter)
+                    ).filter(~col(column).isin(mapping_to_null))
+
+                    if not turned_to_none_values.isEmpty():
+                        raise mlrun.errors.MLRunInvalidArgumentError(
+                            f"Mapvalues - mapping that changing column type must change all values accordingly"
+                            f" the column! Column- {column}"
+                        )
             else:
                 for val, val_range in column_map["ranges"].items():
                     min_val = val_range[0] if val_range[0] != "-inf" else -np.inf
                     max_val = val_range[1] if val_range[1] != "inf" else np.inf
                     otherwise = ""
-                    if new_column_name in event.columns:
-                        otherwise = event[new_column_name]
-                    event = event.withColumn(
+                    if new_column_name in df.columns:
+                        otherwise = df[new_column_name]
+                    df = df.withColumn(
                         new_column_name,
                         when(
-                            (event[column] < max_val) & (event[column] >= min_val),
+                            (df[column] < max_val) & (df[column] >= min_val),
                             lit(val),
                         ).otherwise(otherwise),
                     )
 
         if not self.with_original_features:
-            event = event.select(*self.mapping.keys())
+            df = df.select(*self.mapping.keys())
 
-        return event
+        return df
+
+    @classmethod
+    def validate_args(cls, feature_set, **kwargs):
+        mapping = kwargs.get("mapping", [])
+        for column, column_map in mapping.items():
+            types = set(
+                type(i)
+                for i in column_map.values()
+                if type(i) not in (None, float("nan"), np.nan)
```

Comment:
This condition doesn't work (and none of the test caught it).
```python
>>> float("nan") in [float("nan")]
False
```
This is because NaN != NaN, and set membership behaves in accordance with equality.

Reply:
I think you need to change this to
```suggestion
                type(val)
                for val in column_map.values()
                if type(val) is not None and not math.isnan(val)
```